src/schema/filter_artworks.ts

```diff
@@ -103,38 +103,58 @@ export const FilterArtworksType = new GraphQLObjectType<any, ResolverContext>({
       resolve: (
         { options: gravityOptions },
         args,
-        { filterArtworksLoader }
+        {
+          unauthenticatedLoaders: { filterArtworksLoader: loaderWithCache },
+          authenticatedLoaders: { filterArtworksLoader: loaderWithoutCache },
+        }
       ) => {
         const relayOptions = convertConnectionArgsToGravityArgs(args)
+        const {
+          include_artworks_by_followed_artists,
+          aggregations,
+        } = gravityOptions
+        const requestedPersonalizedAggregation = aggregations.includes(
+          "followed_artists"
+        )
 
-        return filterArtworksLoader(
-          assign(gravityOptions, relayOptions, {})
-        ).then(({ aggregations, hits }) => {
-          if (!aggregations || !aggregations.total) {
-            throw new Error("This query must contain the total aggregation")
-          }
+        let loader
+        if (
+          include_artworks_by_followed_artists ||
+          requestedPersonalizedAggregation
+        ) {
+          loader = loaderWithoutCache
+        } else {
+          loader = loaderWithCache
```

Comment:
This branch assumes that the loader isn’t `undefined`, which could well be the case if the user isn’t actually authenticated. Return `null` instead.

Reply:
True, that would be a case of an unauthenticated user adding params that make this an authenticated request. That can be handled better.

Reply:
I went with a more helpful error instead of just resolving null.

Reply:
Cool 👌